diesel/src/sql_types/mod.rs

```diff
@@ -43,6 +43,12 @@ use crate::query_builder::QueryId;
 #[diesel(mysql_type(name = "Tiny"))]
 pub struct Bool;
 
+/// The "char" postgresql type is needed for for e.g. setweight https://www.postgresql.org/docs/current/functions-textsearch.html
+#[doc(hidden)]
+#[derive(Debug, Clone, Copy, Default, QueryId, SqlType)]
+#[diesel(postgres_type(oid = 18, array_oid = 1002))]
+pub struct CChar;
```

Comment:
This type is postgres specific and should be moved here: https://github.com/diesel-rs/diesel/blob/b1bbe7f090cfb25f472a2185204b0c89dd19e8e6/diesel/src/pg/types/mod.rs#L28

In addition this type misses a list of built-in `FromSql`/`ToSql` impls as it is given for any of the other types.

Reply:
Which `FromSql`/`ToSql` impls are missing? I thought the ones https://github.com/diesel-rs/diesel/pull/3343/files#diff-f4f01457883d1d5f5d72d3a001e4cafbdd84de0ee4a45da29b6379ab2d631a6aR24 are sufficient but I'm not familiar with the code base at all.

Reply:
My comment does not refer to actual implementations of `FromSql`/`ToSql`, but only to an explicit list of those implementations in the documentation of this type. See [here](https://github.com/diesel-rs/diesel/blob/b5b8cdb816ce50224494bc3c26910efacd9f2483/diesel/src/sql_types/mod.rs#L46-L63) for an example how the documentation of similar types is structured. The documentation for this type should follow the same pattern and contain the same information.